containers/daap-python-base/src/var/task/data_platform_paths.py

```diff
@@ -109,6 +114,26 @@ def get_account_id() -> str:
     return boto3.client("sts").get_caller_identity()["Account"]
 
 
+def search_string_for_regex(string: str, regex: re.Pattern[str]) -> Optional[str]:
+    """Search a string for a regex pattern and return the first result"""
+    search_output = regex.search(string)
+    return search_output.groups()[0] if search_output else None
```

Comment:
```suggestion
def search_string_for_regex(string: str, regex: re.Pattern[str]) -> str:
    """Search a string for a regex pattern and return the first result"""
    search_match = regex.search(string)
    if not search_match:
        raise Exception((f"{regex} not found in {string}"))
    return search_match.group(0)
```
Added an error catching for if the regex isn't matched, as `None` isn't handled by downstream uses.
[re.search only matches the first occurrence of the string](https://docs.python.org/3/library/re.html#functions), returning a match with only that group.

Reply:
For my uses case,  if directory structure does match returning None will be useful hence i didnt want to Raise exception

Reply:
I changed it here because None isn't handled in downstream use cases. 

If you want to return None, I'd recommend logging `f"{regex} not found in {string}"` instead of `raise`, then handle `None` in the downstream method calls that at the moment are expecting a string.

Reply:
ok